docs/user/azure/encryption.md

```diff
@@ -0,0 +1,115 @@
+```
+az feature register --namespace "Microsoft.Compute" --name "EncryptionAtHost"
+az feature show --namespace Microsoft.Compute --name EncryptionAtHost
+az provider register -n Microsoft.Compute
+```
+
+### Create resource group
+```
+az group create --name $RESOURCEGROUP --location $LOCATION
+```
+
+### Create key vault
+```
+az keyvault create -n $KEYVAULT_NAME -g $RESOURCEGROUP -l $LOCATION \
+        --enable-purge-protection true --enable-soft-delete true
+
+az keyvault key create --vault-name $KEYVAULT_NAME -n $KEYVAULT_KEY_NAME \
+        --protection software 
+```
+
+### Create an instance of a DiskEncryptionSet
+```
+KEYVAULT_ID=$(az keyvault show --name $KEYVAULT_NAME --query "[id]" -o tsv)
+    
+KEYVAULT_KEY_URL=$(az keyvault key show --vault-name $KEYVAULT_NAME --name \
+        $KEYVAULT_KEY_NAME --query "[key.kid]" -o tsv)
+    
+az disk-encryption-set create -n $DISK_ENCRYPTION_SET_NAME -l $LOCATION -g \
+        $RESOURCEGROUP --source-vault $KEYVAULT_ID --key-url $KEYVAULT_KEY_URL
+```
+
+### Grant the DiskEncryptionSet resource access to the key vault
+```
+DES_IDENTITY=$(az disk-encryption-set show -n $DISK_ENCRYPTION_SET_NAME -g \
+        $RESOURCEGROUP --query "[identity.principalId]" -o tsv)
+
+az keyvault set-policy -n $KEYVAULT_NAME -g $RESOURCEGROUP --object-id \
+        $DES_IDENTITY --key-permissions wrapkey unwrapkey get
+```
+
+### Grant service principal reader permissions to the DiskEncryptionSet
+```
+DES_RESOURCE_ID=$(az disk-encryption-set show -n $DISK_ENCRYPTION_SET_NAME -g \
+        $RESOURCEGROUP --query "[id]" -o tsv)
+
+az role assignment create --assignee $CLUSTER_SP_ID --role Owner --scope \
+        $DES_RESOURCE_ID -o jsonc
+```
+
+### Configure the disk encryption set in your install-config.yaml
+```yaml
+apiVersion: v1
+baseDomain: my.basedomain.com
+compute:
+- architecture: amd64
+  hyperthreading: Enabled
+  name: worker
+  platform:
+    azure:
+      encryptionAtHost: true
+      osDisk:
+        diskEncryptionSet:
+          resourceGroup: diskEncryptionResourceGroup
+          name: diskEncryptionSetName
+  replicas: 3
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform:
+    azure:
+      encryptionAtHost: true
+      osDisk:
+        diskEncryptionSet:
+          resourceGroup: diskEncryptionResourceGroup
+          name: diskEncryptionSetName
+  replicas: 3
+...
+```
+
+### Run the installer
+```
+openshift-install create cluster --log-level debug
+```
+
+
+### Update the annotations on the default storage class
+```
+oc patch storageclass managed-premium -p '{"metadata": # {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}'
+```
+
+### Grab the identity of resource group created by the installer
+```
+az identity list --resource-group "<Cluster Resource Group>"
+```
+
+### Grant service principal write permissions to the DiskEncryptionSet
+```
+az role assignment create --role Owner --assignee "<Identity Name>"
```

Comment:
hm is this set of permissions very broad? That would work fine for dev purposes, but users may balk. Perhaps we could add a sentence before this command that says: "The cluster's resource group needs permissions to write to the disk encryption set. You could, for example, make the resource group identity an `Owner.`"

Reply:
I'd say it is broad for the identity name. Stricter permissions are probably possible, but after days of trying to get this to work, I stopped when it worked ;-) I'm happy to leave it to the admins to use stricter permissions. I've updated the document with your words.